src/app/containers/ArticleTimestamp/helpers.test.js

```diff
@@ -1,20 +1,137 @@
-import { isValidDateTime } from './helpers';
+import {
+  isValidDateTime,
+  isFirstRelative,
+  isLastRelative,
+  isSameDay,
+  isToday,
+  formatType,
+} from './helpers';
 
-const timestamp = 1539969006000; // 19 October 2018
+import { timestampGenerator } from './testHelpers';
+
+const generateTimeLimits = n => {
+  const limits = [];
+  for (let i = 1; i < n; i += 1) {
+    const expectation = {
+      title: `${i} hour(s) ago`,
+      time: timestampGenerator({ hours: i }),
+      expectedValue: i < 10,
+    };
+    limits.push(expectation);
+  }
+  return limits;
+};
 
 describe('ArticleTimestamp helper functions', () => {
   describe('isValidDateTime', () => {
     it('should return true if timestamp is valid', () => {
-      expect(isValidDateTime(timestamp)).toEqual(true);
-      expect(isValidDateTime(0)).toEqual(true);
-      expect(isValidDateTime(-30000000)).toEqual(true);
+      const timestamp = 1539969006000; // 19 October 2018
+
+      expect(isValidDateTime(timestamp)).toBe(true);
+      expect(isValidDateTime(0)).toBe(true);
+      expect(isValidDateTime(-30000000)).toBe(true);
     });
 
     it('should return false if timestamp is invalid or missing', () => {
-      expect(isValidDateTime('foo')).toEqual(false);
-      expect(isValidDateTime(null)).toEqual(false);
-      expect(isValidDateTime(undefined)).toEqual(false);
-      expect(isValidDateTime()).toEqual(false);
+      expect(isValidDateTime('foo')).toBe(false);
+      expect(isValidDateTime(null)).toBe(false);
+      expect(isValidDateTime(undefined)).toBe(false);
+      expect(isValidDateTime()).toBe(false);
+      expect(isValidDateTime(NaN)).toBe(false);
+    });
+  });
+
+  describe('isFirstRelative', () => {
+    describe(`when lastPublished === firstPublished`, () => {
+      const timeLimits = generateTimeLimits(15);
```

Comment:
The helper is a good idea, and the output is very readable. But I think doing 1-15h is probably overkill and produces too much output. Let's just test 1h, 9h and 10h as the most important boundaries.